### PR TITLE
First assignment: Balanced Brackets

### PR DESCRIPTION
**Pull request by klarakufova**
Creating a pull request. No tests yet, just the code.

### CODE REVIEW

**Comment by klarakufova** [comment]
Tests added.

File: src/test/java/com/mycompany/hackerrank/DecideIfCorrectTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            assertFalse(BalancedBrackets.decideIfCorrect(stringNotDivisibleByTwo));
+        } catch (IllegalArgumentException e) {
+            assertEquals("The input does not match the constraints!",e.getMessage());
+        }
+    }
+    
+    @Test    
+    public void testEmptyString() throws Exception {
+        try {
+            BalancedBrackets.decideIfCorrect(emptyString);
```
Comment (rsynek) [inline src/test/java/com/mycompany/hackerrank/DecideIfCorrectTest.java]: What if the method would not throw any exception?

File: src/test/java/com/mycompany/hackerrank/DecideIfCorrectTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        } catch (IllegalArgumentException e) {
+            assertEquals("The input does not match the constraints!",e.getMessage());
+        }
+    }
+    
+    @Test    
+    public void testTooLongString() throws Exception {
+        try {
+            BalancedBrackets.decideIfCorrect(generateLongString());
```
Comment (rsynek) [inline src/test/java/com/mycompany/hackerrank/DecideIfCorrectTest.java]: The same as above and there are several more occurrences below.

**Comment by rsynek** [comment]
Several comments inline; otherwise looks good. Nice ideas about testing negative scenarios.

Thanks

**Comment by klarakufova** [comment]
@rsynek, all your notes incorporated. Can we merge the pull request?

File: src/test/java/com/mycompany/hackerrank/DecideIfCorrectTest.java (hunk 1 above)
Comment (rsynek) [inline src/test/java/com/mycompany/hackerrank/DecideIfCorrectTest.java]: I would rather use http://junit.sourceforge.net/javadoc/org/junit/Assert.html#fail(java.lang.String) method - now you don't now what went wrong until you take a look in the test source code.

**Comment by rsynek** [comment]
One more comment regarding the `fail` method - see in the src.

**Comment by klarakufova** [comment]
@rsynek, that's a very good point! Incorporated.

**Comment by rsynek** [comment]
Very good, thanks!

**Comment by rsynek** [comment]
@kkufova one more thing; could you please squash/rebase your work to a single commit?

**Pull request merged**